torchdynamo/variables/misc.py

```diff
@@ -35,10 +36,13 @@ def const_getattr(self, tx, name):
         assert self.objvar, "1-arg super not implemented"
         search_type = self.typevar.as_python_constant()
 
-        # We default to the python type of the object. However, if this is a
-        # `type`, then the original object represents the user defined type.
+        # We default to the python type of the object. However,
+        # 1. If this is a `type`, then the original object represents the user
+        # defined type.
+        # 2. If this is `torch._C._TensorMeta`, the original object is the user
+        # defined type of a custom tensor subclass.
         type_to_use = self.objvar.python_type()
-        if type_to_use is type:
+        if type_to_use is type or type_to_use is torch._C._TensorMeta:
```

Comment:
I wonder if there is a better way to handle this, as it feels a bit like a hack to do different things based on the value.  Perhaps we could detect we are in a `@classmethod` and do something different.  Or look at the args to `super()` and do this logic based on if `super()` got a type or an object.  Ideally this would mirror what cpython does.